File: 1961.check-if-string-is-a-prefix-of-array.cpp
/*
 * @lc app=leetcode id=1961 lang=cpp
 *
 * [1961] Check If String Is a Prefix of Array
 */
#include <string>
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    bool isPrefixString(string s, vector<string> &words)
    {
        int width = s.size();
        int ind = 0;
        for (auto &&word : words)
        {
            for (auto c : word)
            {
                if (ind == width || c != s[ind])
                {
                    return false;
                }
                else
                {
                    ++ind;
                }
            }
            if (ind == width)
            {
                return true;
            }
        }
        return false;
    }
};
// @lc code=end

File: 1961.check-if-string-is-a-prefix-of-array.py
#
# @lc app=leetcode id=1961 lang=python3
#
# [1961] Check If String Is a Prefix of Array
#

# @lc code=start
class Solution:
    def isPrefixString(self, s: str, words: List[str]) -> bool:
        width = len(s)
        ind = 0
        for word in words:
            for c in word:
                if ind == width or c != s[ind]:
                    return False
                else:
                    ind += 1
            if ind == width:
                return True
        return False
# @lc code=end
